src/ensure_buffer.ts

```diff
@@ -11,6 +11,11 @@ import { isAnyArrayBuffer } from './parser/utils';
  * @throws BSONTypeError If anything other than a Buffer or Uint8Array is passed in
  */
 export function ensureBuffer(potentialBuffer: Buffer | ArrayBufferView | ArrayBuffer): Buffer {
+  if (potentialBuffer instanceof Buffer) {
```

Comment:
Why is it an instanceof check?

Imho should be 

```suggestion
  if (Buffer.isBuffer(potentialBuffer)) {
```

Reply:
Also js-bson uses npm package buffer, which also provides Buffer.isBuffer.

Reply:
Hey @Uzlopak we chose to use `instanceof` here because of it's sensitivity to buffers passed from other environments. In nodejs this will always return true so we can optimize for that case. In browsers if a different instance is passed to use we can be sure we align it with the type and version of buffer that our library expects.

Reply:
Can you please elaborate the Browser case? I mean instanceof would return true only if it is an instance of the buffer class which is used by the js-bson package. So if you would pass a buffer of a different class, it would also fail. Or am I wrong?

Reply:
Sure yea, it would return false for other implementations of the class which is the desired behavior because it will wrap the passed buffer in the exact version we depend on. It is possible for other versions of the `Buffer` polyfill to be included in the same bundle or `Buffer`s from across Realm boundaries to be passed to our APIs.
Here we want to make a minimal change the alleviates a performance cost on node that isn't necessary since it's the expected type already and there's no need to wrap the type.